### ISSUE TITLE
vec_as_names repair = "universal" can be non-unique

### ISSUE THREAD

**Issue by dpritchard** [r-lib/vctrs#1013]
It is possible to receive non-unique names from `vec_as_names(..., repair = "universal")`as follows:

````r
> vec_as_names(c("", "1"), repair = "universal", quiet = TRUE)
[1] "...1" "...1"
````

This could happen wherever empty strings and character representations of integers co-exist and the 'integer' corresponds to the position of the empty string. 

````r
> vec_as_names(c("", "5", "a", "b", "", "1"), repair = "universal", quiet = TRUE)
[1] "...1" "...5" "a"    "b"    "...5" "...1"
````

The issue is that in the hierarchy of `minimal` < `unique` < `universal` the logic of making character representations of integers `universal` undermines the logic applied to make empty strings `unique`. Everything is fine at the `unique`-only stage:

````r
> vec_as_names(c("", "1"), repair = "unique", quiet = TRUE)
[1] "...1" "1"
````

Apologies if this is known and I have just missed it. But at the least this seems counter to the docs ("`universal` names are `unique` and syntactic...") and doesn't seem to be mentioned as a known issue in the [design guide](https://principles.tidyverse.org/names-attribute.html). At the least, this seems worthy of a mention as an edge case?

**Comment by dpritchard** [comment]
Also, FWIW

````r
> vctrs::vec_as_names(c("..1", "1"), repair = "universal", quiet = TRUE)
[1] "...1" "...1"
````

**Comment by jennybc** [comment]
@dpritchard Did you encounter this in actual usage?

I will need to go back to the original implementation, which happened in tibble and got re-implemented here, to see if we've always had this problem.

**Comment by dpritchard** [comment]
> Did you encounter this in actual usage?

Yes and no... I needed a consistent approach to naming in our own (internal) package. I had planned to leverage the 'tidyverse approach' and just stumbled on to this when writing tests for the package. 

Sorry! It was simply the "non-uniqueness" caught my attention initially. What (if anything at all!) to do about it might be a harder nut to crack.
